pkg/ingester/ingester.go

```diff
@@ -2079,10 +2080,35 @@ func wrappedTSDBIngestErr(ingestErr error, timestamp model.Time, labels []mimirp
 		return nil
 	}
 
-	return fmt.Errorf(errTSDBIngest, ingestErr, timestamp.Time().UTC().Format(time.RFC3339Nano), mimirpb.FromLabelAdaptersToLabels(labels).String())
+	var errID globalerror.ID
+	var errMsg string
+	switch cause := errors.Cause(ingestErr); cause {
+	case storage.ErrOutOfBounds:
+		errID = globalerror.OutOfBounds
+		errMsg = "something out of bounds"
+	case storage.ErrOutOfOrderSample:
+		errID = globalerror.OutOfOrderSample
+		errMsg = "something sample out of order"
```

Comment:
Error message could be something like "the sample has been rejected because another sample with a more recent timestamp has already been ingested and out of order samples are not allowed"